### PR TITLE
Can add ssh

### PR DESCRIPTION
**Pull request by wangcan0329**
docker container support ssh

### CODE REVIEW

File: rest-server/src/templates/dockerContainerScript.mustache [hunk 1]
```diff
@@ -52,7 +52,25 @@ export PAI_JOB_TASK_ROLE_COUNT={{{ taskRolesNumber }}}
 export PAI_JOB_TASK_ROLE_LIST={{{ taskRoleList }}}
 export PAI_KILL_ALL_ON_COMPLETED_TASK_NUM={{{ jobData.killAllOnCompletedTaskNumber }}}
 
-export PAI_CONTAINER_HOST_PORT=$(( (RANDOM % 55535) + 10001 ))
+# generate container host port
+PAI_CONTAINER_HOST_PORT=$(( (RANDOM % 55535) + 10001 ))
```
Comment (hwuu) [inline rest-server/src/templates/dockerContainerScript.mustache]: Can you write `PAI_CONTAINER_HOST_PORT=$(( (RANDOM % 55535) + 10001 ))` once? For example:
```
while true
  CONTAINER_HOST_PORT=...
  if not lsof -i:${PAI_CONTAINER_SSH_PORT}
    break
  else
    sleep 1
```
In this way, we don't have to change two lines if we need to modify the port range in the future.

File: rest-server/src/templates/dockerContainerScript.mustache [hunk 2]
```diff
@@ -105,6 +123,73 @@ export PAI_TASK_ROLE_LIST=$PAI_JOB_TASK_ROLE_LIST
...
+      "[INFO]" "begin to download ssh key pair from hdfs ..."
+  hdfs dfs -get "${hdfs_ssh_folder}/.ssh/" "/root/" || exit 1
+fi
+# Start ssh service
+start_ssh_service
+# Generate ssh connect info file in "PAI_CONTAINER_ID-PAI_CURRENT_CONTAINER_IP-PAI_CONTAINER_SSH_PORT" format on hdfs
+hdfs dfs -touchz ${hdfs_ssh_folder}/$PAI_CONTAINER_ID-$PAI_CONTAINER_HOST_IP-$PAI_CONTAINER_SSH_PORT || exit 1
+
+# Write env to bashrc
+env | sed "s/^/export /" >> ~/.bashrc
```
Comment (abuccts) [inline rest-server/src/templates/dockerContainerScript.mustache]: `grep "^PAI"` before `sed`?

File: rest-server/src/templates/dockerContainerScript.mustache [hunk 3]
```diff
@@ -52,7 +52,25 @@ export PAI_JOB_TASK_ROLE_COUNT={{{ taskRolesNumber }}}
 export PAI_JOB_TASK_ROLE_LIST={{{ taskRoleList }}}
 export PAI_KILL_ALL_ON_COMPLETED_TASK_NUM={{{ jobData.killAllOnCompletedTaskNumber }}}
 
-export PAI_CONTAINER_HOST_PORT=$(( (RANDOM % 55535) + 10001 ))
+# generate container host port
```
Comment (hwuu) [inline rest-server/src/templates/dockerContainerScript.mustache]: `generate` --> `Determine the`

File: rest-server/src/templates/dockerContainerScript.mustache (hunk 2 above)
Comment (wangcan0329) [reply rest-server/src/templates/dockerContainerScript.mustache]: I discussed with Bin, if we only keep environment started with "PAI", some configurations like hdfs will be incorrect when SSH connect into container.

Comment (abuccts) [reply rest-server/src/templates/dockerContainerScript.mustache]: ok

**Pull request merged**